src/routes/sns/+page.svelte

```diff
@@ -0,0 +1,516 @@
+<script lang="ts">
+	import StatsWidget from '../stake/StatsWidget.svelte';
+	import SnsListing from './SnsListing.svelte';
+	import CopyIcon from '$lib/icons/CopyIcon.svelte';
+	import { fade, scale } from 'svelte/transition';
+	import { afterUpdate, onMount } from 'svelte';
+	import { snsPrincipal, selectedSns, canisters, isBusy, toasts, snsHex } from '$lib/stores';
+	import { Toast } from '$lib/toast';
+	import { handleSnsIcpDepositResult, handleSnsRetrieveNicpResult } from '$lib/resultHandler';
+	import { Principal } from '@dfinity/principal';
+	import { type Account, SubAccount, AccountIdentifier } from '@dfinity/ledger-icp';
+	import BigNumber from 'bignumber.js';
+	import { displayUsFormat, bigintE8sToNumber } from '$lib';
+	import { signIn, CANISTER_ID_BOOMERANG } from '$lib/authentification';
+	import { fetchIcpBalance, fetchNicpBalance } from '$lib/state';
+	import {
+		uint8ArrayToHexString,
+		hexStringToUint8Array,
+		bigEndianCrc32,
+		encodeBase32
+	} from '@dfinity/utils';
+	import { decodeIcrcAccount } from '@dfinity/ledger-icrc';
+
+	let icpBalance: BigNumber;
+	let nicpBalance: BigNumber;
+	let previousHex: string;
+	let isConfirmBusy: boolean;
+	let isRetrieveBusy: boolean;
+
+	function isPrincipalValid(input: string): boolean {
+		try {
+			Principal.fromText(input);
+			return true;
+		} catch (error) {
+			return false;
+		}
+	}
+
+	const setAccount = async () => {
+		if ($canisters && isPrincipalValid($snsPrincipal)) {
+			try {
+				const account = await $canisters.boomerang.get_staking_account(
+					Principal.fromText($snsPrincipal)
+				);
+
+				const crc = bigEndianCrc32(
+					Uint8Array.from([...account.owner.toUint8Array(), ...account.subaccount[0]])
+				);
+				const crc32 = encodeBase32(crc);
+
+				const hex =
+					CANISTER_ID_BOOMERANG + '-' + crc32 + '.' + uint8ArrayToHexString(account.subaccount[0]);
+
+				if (hex !== previousHex) {
+					snsHex.set(hex);
+					previousHex = hex;
```

Comment:
All of that can be done with: https://www.npmjs.com/package/@dfinity/ledger-icrc#gear-encodeicrcaccount

Reply:
done.